File: luna-client/src/models/workspace/WorkspaceView.ts
import { WorkspaceUserView } from "@/models/workspace/WorkspaceUserView";

export interface WorkspaceView {
	id: string;
	name: string;
	ownerId: string;
	icon?: string;
	description?: string;
	defaultPermission: string;
	settings? : object;
	deletedAt?: string;
	users: WorkspaceUserView[];
}

File: Luna.Workspaces.Models.View/Models/WorkspaceView.cs
namespace Luna.Workspaces.Models.View.Models;

public class WorkspaceView
{
	public Guid Id { get; set; }
	public string Name { get; set; } = null!;
	public Guid OwnerId { get; set; }
	public string? Icon { get; set; }
	public string? Description { get; set; }
	public string DefaultPermission { get; set; } = null!;
	public object? Settings { get; set; }
	public DateTime? DeletedAt { get; set; }
	public IEnumerable<WorkspaceUserView> Users { get; set; } = new List<WorkspaceUserView>();
}